File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ErrorEvent.java
package work.socialhub.kslack.entity.event;

/**
 * If there was a problem connecting an error will be returned,
 * including a descriptive error message:
 *
 * <pre>
 * {
 *     "type": "error",
 *     "error": {
 *         "code": 1,
 *         "msg": "Socket URL has expired"
 *     }
 * }
 * </pre>
 * <p>
 * https://api.slack.com/rtm
 */
public class ErrorEvent implements Event {

    public static final String TYPE_NAME = "error";

    private final String type = TYPE_NAME;
    private Error error;

    public String getType() {
        return this.type;
    }

    public Error getError() {
        return this.error;
    }

    public void setError(Error error) {
        this.error = error;
    }

    public static class Error {
        private Integer code;
        private String msg;

        public Integer getCode() {
            return this.code;
        }

        public String getMsg() {
            return this.msg;
        }

        public void setCode(Integer code) {
            this.code = code;
        }

        public void setMsg(String msg) {
            this.msg = msg;
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/ErrorEvent.kt
package work.socialhub.kslack.entity.event

/**
 * If there was a problem connecting an error will be returned,
 * including a descriptive error message:
 *
 * <pre>
 * {
 * "type": "error",
 * "error": {
 * "code": 1,
 * "msg": "Socket URL has expired"
 * }
 * }
</pre> *
 *
 *
 * https://api.slack.com/rtm
 */
class ErrorEvent : Event {
    override val type: String = TYPE_NAME
    var error: Error? = null

    class Error {
        var code: Int? = null
        var msg: String? = null
    }

    companion object {
        const val TYPE_NAME: String = "error"
    }
}